### ISSUE TITLE
[Plot] Adopt scripting mechanism to automate plots and store in project.

### ISSUE THREAD

**Issue by sky5walk** [sqlitebrowser/sqlitebrowser#1697]
I find the existing plot feature is a bit untapped in its potential. Tucked away in the corner, and always on top when peeled away?
I propose the creation or adoption of an existing scripting mechanism to automate the plot features.
We have 'Execute SQL', please add 'Execute PLOT' and give it its own tab on the main panel.

Then we can 'play' our SQL queries as graphical dumps to corresponding PLOT windows.
Parameters like X,Y,Colors, grid, Title, etc. can be captured in the script.

**Comment by justinclift** [comment]
This sounds like an interesting concept.  It's also in line with a weird idea (not documented) I had ages ago of maybe integrating some kind of extended analysis capability.

Haven't actually looked into whether integrating (say GNU Octave) or embedded (say) R is even feasible (OSS licensing, cross platform code, etc) though.  Thus far, it's just been an idea in my head. :smile:

**Comment by sky5walk** [comment]
Yes, those packages are quite extensive. Stripped down versions of the plotting libs would be very useful. If this avoids recreating a wheel?
